packages/beacon-node/src/chain/chain.ts

```diff
@@ -683,11 +687,21 @@ export class BeaconChain implements IBeaconChain {
       });
     }
 
-    // Prune old blobsSidecar for block production, those are only useful on their slot
-    if (this.config.getForkSeq(slot) >= ForkSeq.deneb && this.producedBlobsSidecarCache.size > 0) {
-      for (const [key, blobsSidecar] of this.producedBlobsSidecarCache) {
-        if (slot > blobsSidecar.beaconBlockSlot + MAX_RETAINED_SLOTS_CACHED_BLOBS_SIDECAR) {
-          this.producedBlobsSidecarCache.delete(key);
+    // Prune old blobSidecars for block production, those are only useful on their slot
+    if (this.config.getForkSeq(slot) >= ForkSeq.deneb) {
+      if (this.producedBlobSidecarsCache.size > 0) {
+        for (const [key, {slot: blobSlot}] of this.producedBlobSidecarsCache) {
```

Comment:
not related to this PR, why don't we have `producedBlobSidecarsCache` with slot or block number as key so that we don't have to do a full scan here? I think we should have slot/block number in all of the consumer of the cache

Reply:
because its not re-org safe

Reply:
I mean we cache by slot, and then by block hash, and when we search we filter by slot first, then by block hash (I notice all of the consumers have slot).
it'd be easier to prune by slot

Reply:
hmmm can be done but sounds like an overkill since the freshest blobs are the most likely to be used in the proposal and are added to cache by increasing slot order, so pruning out by max naturally follows that order